src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -379,18 +386,23 @@ static void clinit(ClassWriter cw, String className, List<ClassData> classData)
         mv.visitLdcInsn(Type.getType("L" + className + ";"));
         mv.visitMethodInsn(Opcodes.INVOKESTATIC, "java/lang/invoke/MethodHandles",
                            "classData", "(Ljava/lang/Class;)Ljava/lang/Object;", false);
-        // we should optimize one single element case that does not need to create a List
-        mv.visitTypeInsn(Opcodes.CHECKCAST, "java/util/List");
-        mv.visitVarInsn(Opcodes.ASTORE, 0);
-        int index = 0;
-        for (ClassData p : classData) {
-            // initialize the static field
-            mv.visitVarInsn(Opcodes.ALOAD, 0);
-            emitIconstInsn(mv, index++);
-            mv.visitMethodInsn(Opcodes.INVOKEINTERFACE, "java/util/List",
-                               "get", "(I)Ljava/lang/Object;", true);
+        if (classData.size() == 1) {
```

Comment:
```
if (classData.size() < 2) {
```
It works for zero case, because `checkcast` always succeeds on nulls.

Reply:
Actually the zero case is already handled here:

```
    static void clinit(ClassWriter cw, String className, List<ClassData> classData) {
        if (classData.isEmpty())
            return;
```

No `<clinit>` method is generated because we have no static field to initialize.